skt/executable.py

```diff
@@ -221,70 +231,69 @@ def cmd_merge(cfg):
                                        merge_report_path, report_string)
 
                     return
-
                 report_string += '\n  - %s\n    into commit %s' % (mb[0],
                                                                    bhead[:12])
-                idx += 1
+                idx[0] += 1
 
-        elif cfg.get('patch') or cfg.get('pw'):
-            report_string = 'We applied the following patch(es):'
+            elif thing_to_merge[0] == 'patch' or thing_to_merge[0] == 'pw':
+                report_string += '\nWe applied the following patch(es):'
 
-            if cfg.get('patch'):
-                utypes.append("[local patch]")
-                for patch in cfg.get('patch'):
+                if thing_to_merge[0] == 'patch':
+                    utypes.append("[local patch]")
```

Comment:
Putting it here means the type will be added with each patch applied. So if we grab series with 50 patches, the type will be present in the state file 50 times, which is not exactly wanted. Since you went with the array of indices, it may be better to iterate over it at the end and add the type for each non-zero one, what do you think?

Reply:
@veruu yeah it is right. I can add it at the end. Or I can add it at beginning (outside the loop). I think the way to fix the string at the beginning and, add the type at the same time.